XcodePostBuild.cs

```diff
@@ -519,4 +680,154 @@ private static void EditCodeFile(string path, Func<string, IEnumerable<string>>
+    /// <param name="fullPath">Full path.</param>
+    /// <param name="directory">The directory from which to consider the relative path. If no value 
+    /// is provided (i.e. `null` or empty string), then the current working directory is used.</param>
+    public static string Abs2Rel(string fullPath, string directory = null)
+    {
+        if (!Path.IsPathRooted(fullPath))
+        {
+            return fullPath;
+        }
+        else
+        {
+            if (string.IsNullOrEmpty(directory))
+            {
+                directory = Environment.CurrentDirectory;
+            }
+
+            var partsA = directory.Split('/', '\\').ToList();
+            var partsB = fullPath.Split('/', '\\').ToList();
+
+            while (partsA.Count > 0
+                   && partsB.Count > 0
+                   && partsA[0] == partsB[0])
+            {
+                partsA.RemoveAt(0);
+                partsB.RemoveAt(0);
+            }
+
+            if (partsB.Count == 0)
+            {
+                return null;
+            }
+            else
+            {
+                return Combine(partsA
+                    .Select(_ => "..")
+                    .Concat(partsB)
+                    .ToArray());
+            }
+        }
+    }
+
+    /// <summary>
+    /// Resolves an absolute path from a path that is relative to the currently-edited
+    /// demo path.
+    /// </summary>
+    /// <returns>The absolute path.</returns>
+    /// <param name="relativePath">Relative path.</param>
+    /// <param name="directory">The directory from which to consider the relative path. If no value 
+    /// is provided (i.e. `null` or empty string), then the current working directory is used.</param>
+    public static string Rel2Abs(string relativePath, string directory = null)
+    {
+        if (Path.IsPathRooted(relativePath))
+        {
+            return relativePath;
+        }
+        else
+        {
+            if (string.IsNullOrEmpty(directory))
+            {
+                directory = Environment.CurrentDirectory;
+            }
+
+            var partsA = directory.Split('/', '\\').ToList();
+            var partsB = relativePath.Split('/', '\\').ToList();
+
+            while (partsA.Count > 0
+                   && partsB.Count > 0
+                   && partsB[0] == "..")
+            {
+                partsA.RemoveAt(partsA.Count - 1);
+                partsB.RemoveAt(0);
+            }
+
+            if (partsB.Count == 0)
+            {
+                return null;
+            }
+            else
+            {
+                return Combine(partsA
+                    .Concat(partsB)
+                    .ToArray());
+            }
+        }
+    }
+
+    /// <summary>
+    /// Fills in directory structures as necessary to make sure that file operations on
+    /// `path` will succeed.
+    /// </summary>
+    /// <param name="path"></param>
+    public static void FillDirectories(string path)
```

Comment:
`Directory.CreateDirectory` will create all intermediate directories if not already exists.  So this can be simply:
```
Directory.CreateDirectory(Path.GetDirectoryname(path));
```

https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=netframework-3.0#System_IO_Directory_CreateDirectory_System_String_

Reply:
Ah, cripes, don't know how I missed that